### PR TITLE
Patcher GUI

### LINKED ISSUE #5: Feature: GUI for patcher
The patcher should have a GUI; this can either be a canvas like what Pixel gives us, or a simpler GUI window with text and buttons to explain to the user what's happening while the patcher runs (as well as to display any errors the patcher might encounter).

### PR DESCRIPTION
**Pull request by aerth**
Simple GUI for patcher, should close #5

Adds `popup` package for client to use gxui elements for notifications etc

### CODE REVIEW

File: patcher/main.go [hunk 1]
```diff
@@ -123,7 +127,11 @@ func downloadClient(clientName string) error {
 		log.Printf("skipping download")
 		return nil
 	}
-	log.Printf("downloading client")
+
+	if res.StatusCode != http.StatusOK {
+		return fmt.Errorf("got 404")
```
Comment (ilackarms) [inline patcher/main.go]: maybe we should give the actual status code? it might not be a 404

File: patcher/main.go [hunk 2]
```diff
@@ -26,20 +26,10 @@ var protocol = flag.String("protocol", "udp", fmt.Sprintf("network protocol to u
 
 func main() {
 	flag.Parse()
-
-	logFile, err := os.Create("game.log")
```
Comment (ilackarms) [inline patcher/main.go:30]: why remove this?

File: patcher/gui.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+	window.AddChild(splitter)
+	window.OnClose(driver.Terminate)
+
+}
+
+func wrap(s string) string {
+	var out string
+	for i, c := range s {
+		if i%4 == 0 {
```
Comment (ilackarms) [inline patcher/gui.go]: maybe 4 should be an argument (i.e. `width`)

File: patcher/main.go (hunk 2 above)
Comment (aerth) [reply patcher/main.go:30]: Changed from os.Create to os.Open, using APPEND and CREATE mode as to not overwrite existing log

While patcher is open, errors should be in the gxui

Comment (aerth) [inline patcher/main.go]: Moved back to main(), using global io.Writer `out`

File: patcher/main.go [hunk 4]
```diff
@@ -23,24 +23,22 @@ var addr = flag.String("addr", "localhost:8080", "http service address")
...
-	out := io.MultiWriter(logFile, os.Stdout)
-
-	logger := log.New(out, "", log.LstdFlags)
-
 	if *playerID == "" {
-		confData, err := ioutil.ReadFile(*confFile)
-		if err != nil {
-			logger.Fatalf("%s not found: %v", *confFile, err)
-		}
+		*playerID = uuid.New()
```
Comment (aerth) [inline patcher/main.go]: Generates random uuid if no config file at specified location

**Comment by ilackarms** [comment]
is this PR ready for review?

**Pull request closed without merging**